homeassistant/components/shopping_list/__init__.py

```diff
@@ -86,6 +86,7 @@ async def add_item_service(call: ServiceCall) -> None:
         data = hass.data[DOMAIN]
         if (name := call.data.get(ATTR_NAME)) is not None:
             await data.async_add(name)
+            hass.bus.async_fire(EVENT)
```

Comment:
Can we move this to `ShoppingData.async_add` instead? Same comments for the other services.

Reply:
Thanks for the review. I moved all async_fire calls to the data class.